dotnet/src/Connectors/Connectors.AzureOpenAI/Core/ClientCore.cs

```diff
@@ -92,674 +52,85 @@ internal ClientCore(ILogger? logger = null)
+    /// Initializes a new instance of the <see cref="ClientCore"/> class.
     /// </summary>
-    private static readonly Counter<int> s_promptTokensCounter =
-        s_meter.CreateCounter<int>(
-            name: "semantic_kernel.connectors.openai.tokens.prompt",
-            unit: "{token}",
-            description: "Number of prompt tokens used");
-
-    /// <summary>
-    /// Instance of <see cref="Counter{T}"/> to keep track of the number of completion tokens used.
-    /// </summary>
-    private static readonly Counter<int> s_completionTokensCounter =
-        s_meter.CreateCounter<int>(
-            name: "semantic_kernel.connectors.openai.tokens.completion",
-            unit: "{token}",
-            description: "Number of completion tokens used");
-
-    /// <summary>
-    /// Instance of <see cref="Counter{T}"/> to keep track of the total number of tokens used.
-    /// </summary>
-    private static readonly Counter<int> s_totalTokensCounter =
-        s_meter.CreateCounter<int>(
-            name: "semantic_kernel.connectors.openai.tokens.total",
-            unit: "{token}",
-            description: "Number of tokens used");
-
-    private static Dictionary<string, object?> GetChatCompletionMetadata(OpenAIChatCompletion completions)
+    /// <param name="deploymentName">Azure OpenAI deployment name, see https://learn.microsoft.com/azure/cognitive-services/openai/how-to/create-resource</param>
+    /// <param name="endpoint">Azure OpenAI deployment URL, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
+    /// <param name="apiKey">Azure OpenAI API key, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
+    /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
+    /// <param name="logger">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
+    internal ClientCore(
+        string deploymentName,
+        string endpoint,
+        string apiKey,
+        HttpClient? httpClient = null,
+        ILogger? logger = null)
     {
-#pragma warning disable AOAI001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
-        return new Dictionary<string, object?>(8)
-        {
-            { nameof(completions.Id), completions.Id },
-            { nameof(completions.CreatedAt), completions.CreatedAt },
-            { PromptFilterResultsMetadataKey, completions.GetContentFilterResultForPrompt() },
-            { nameof(completions.SystemFingerprint), completions.SystemFingerprint },
-            { nameof(completions.Usage), completions.Usage },
-            { ContentFilterResultsMetadataKey, completions.GetContentFilterResultForResponse() },
+        Verify.NotNullOrWhiteSpace(deploymentName);
+        Verify.NotNullOrWhiteSpace(endpoint);
```

Comment:
If an endpoint is provided, apiKey may not be required. Should I apply WIB for this too? ...

Reply:
https://github.com/microsoft/semantic-kernel/issues/7065